### PR TITLE
Main- Update to use ArcGIS runtime update 10

### PR DESCRIPTION
**Pull request by puneet-pdx**
- Update versions - 
   Android Gradle Plugin Version - 4.1.2
   Gradle version - 6.5
   Min SDK - 23 required by ArcGIS runtime
   ArcGIS runtime version - 100.10.0
   test runner version
   materialLibraryVersion
   appcompatLibraryVersion

- Update app versionName to 0.1.4

### CODE REVIEW

**Review by gunt0001 — APPROVED** [review]

**Review by changanxian — APPROVED** [review]
👍

**Comment by puneet-pdx** [comment]
@esreli can you please review.

File: README.md [hunk 1]
```diff
@@ -109,13 +109,13 @@ If there are changes made in the original repository, you can sync the fork to k
...
 
-* [Android Studio](http://developer.android.com/sdk/index.html)
+* [Android Studio](https://developer.android.com/studio/)
 
 ## Resources
 
 * [ArcGIS Runtime SDK for Android Developers Site](https://developers.arcgis.com/android/)
-* [ArcGIS Mobile Blog](http://blogs.esri.com/esri/arcgis/category/mobile/)
-* [ArcGIS Developer Blog](http://blogs.esri.com/esri/arcgis/category/developer/)
+* [ArcGIS Apps Blog](https://www.esri.com/arcgis-blog/apps/)
```
Comment (esreli) [inline README.md]: Let's drop this resource, I'm not sure it really applies to us.
```suggestion
```

**Review by esreli — APPROVED** [review]
I think this is looking good 👍 

- All doc links are updated
- App version number is incremented
- Dependencies are updated
- RELEASE notes logged

**Pull request merged**